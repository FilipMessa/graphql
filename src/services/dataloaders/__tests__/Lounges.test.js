// @flow

import LoungesDataloader from '../Lounges';

// intentionally simplified
const mockDatasetPRG = [
  {
    airport: 'Vaclav Havel',
    terminal: 'Terminal 1',
    iata: 'PRG',
    lounge_name: 'Menzies Aviation Lounge',
    lounge_code: 'PRGAHAR',
    provider: 'collinsons',
  },
];

// intentionally simplified
const mockDatasetLHR = [
  {
    airport: 'Heathrow',
    terminal: 'Terminal 2',
    iata: 'LHR',
    lounge_name: 'Plaza Premium Lounge (3-6hr stay)',
    lounge_code: 'LHR2EXP',
    provider: 'collinsons',
  },
  {
    airport: 'London Heathrow',
    terminal: 'Terminal 3',
    iata: 'LHR',
    lounge_name: 'No. 1 Heathrow',
    lounge_code: 'null',
    provider: 'loungebuddy',
  },
];

jest.mock('../../../common/services/JsonFetcher', () => ({
  fetchJson: url => {
    switch (url) {
      case 'https://lounges-api.skypicker.com/lounge?iata=PRG,LHR,unknown':
        return Promise.resolve(mockDatasetPRG.concat(mockDatasetLHR));
      default:
        throw new Error(`No mock data for: ${url}`);
    }
  },
}));

it('loads lounges correctly', async () => {
  // this must be first
  await expect(
    LoungesDataloader.loadMany(['PRG', 'LHR', 'unknown']),
  ).resolves.toMatchSnapshot();

  // these should be already loaded
  await expect(LoungesDataloader.load('PRG')).resolves.toMatchSnapshot();
  await expect(LoungesDataloader.load('LHR')).resolves.toMatchSnapshot();
});
