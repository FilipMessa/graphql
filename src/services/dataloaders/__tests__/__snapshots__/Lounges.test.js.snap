// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads lounges correctly 1`] = `
Array [
  Array [
    Object {
      "airport": "Vaclav Havel",
      "iata": "PRG",
      "loungeCode": "PRGAHAR",
      "loungeName": "Menzies Aviation Lounge",
      "provider": "collinsons",
      "terminal": "Terminal 1",
    },
  ],
  Array [
    Object {
      "airport": "Heathrow",
      "iata": "LHR",
      "loungeCode": "LHR2EXP",
      "loungeName": "Plaza Premium Lounge (3-6hr stay)",
      "provider": "collinsons",
      "terminal": "Terminal 2",
    },
    Object {
      "airport": "London Heathrow",
      "iata": "LHR",
      "loungeCode": "null",
      "loungeName": "No. 1 Heathrow",
      "provider": "loungebuddy",
      "terminal": "Terminal 3",
    },
  ],
  Array [],
]
`;

exports[`loads lounges correctly 2`] = `
Array [
  Object {
    "airport": "Vaclav Havel",
    "iata": "PRG",
    "loungeCode": "PRGAHAR",
    "loungeName": "Menzies Aviation Lounge",
    "provider": "collinsons",
    "terminal": "Terminal 1",
  },
]
`;

exports[`loads lounges correctly 3`] = `
Array [
  Object {
    "airport": "Heathrow",
    "iata": "LHR",
    "loungeCode": "LHR2EXP",
    "loungeName": "Plaza Premium Lounge (3-6hr stay)",
    "provider": "collinsons",
    "terminal": "Terminal 2",
  },
  Object {
    "airport": "London Heathrow",
    "iata": "LHR",
    "loungeCode": "null",
    "loungeName": "No. 1 Heathrow",
    "provider": "loungebuddy",
    "terminal": "Terminal 3",
  },
]
`;
